### ISSUE TITLE
PhysX Force Region Simple Drag launches entities instead of slowing them at certain Mass:Region Density proportions

### ISSUE THREAD

**Issue by LB-ArturZieba** [o3de/o3de#14352]
**Describe the bug**
*PhysX Force Region* component's *Simple Drag* *Force Type* launches up or drags quickly down *PhysX Rigid Body* entity, if proportion of that entity's *Mass* and *Region Density* is at a certain range. For example if the Rigid Body entity has *Mass* of 1kg and *Region Density* is set to 100, that entity will be launched, but at *Region Density* set to 10, that entity will simply fall slower.

It should be noted that this issue can be inconsistent, as on two of the attempts on one of the machines the falling *PhysX Rigid Body* entity was unaffected by *PhysX Force Region* at all. After further testing this did not occur and on other machines this issue had 100% reproduction rate, but we were not able to pinpoint the exact cause of this behavior.

This issue also occurs on PhysX_StaticRigidBody_Staging (50e9c93) (https://github.com/aws-lumberyard-dev/o3de/tree/PhysX_StaticRigidBody_Staging) branch.

**Steps to reproduce**
1. Create "Force Region" entity, add *PhysX Collider* and *PhysX Force Region* (and *PhysX Static Rigid Body* if on PhysX_StaticRigidBody_Staging branch) components to it.
2. Set *PhysX Force Region* *Force Type* to *Simple Drag* and *Region Density* to 100.
3. Enable *PhysX Collider* *Trigger* property, set *Shape* to *Box* and *Dimensions* to for example X = 20; Y = 20; Z = 20.
4. Create "Rigid Body" entity, add *PhysX Collider*, *Mesh* and *PhysX Rigid Body* (or *PhysX Dynamic Rigid Body* if on PhysX_StaticRigidBody_Staging branch) components to it.
5. Disable *PhysX Rigid Body* (or *PhysX Dynamic Rigid Body*) *Compute Mass* property and set *Mass* to 1.
6. Set *PhysX Collider *Shape* to *Box*.
7. Assign a cube mesh asset to the *Mesh* component (for example BeveledCube.fbx).
8. Place "Rigid Body" entity above the "Force Region" entity.
9. Enter the Game Mode.

**Expected behavior**
"Rigid Body" entity falls slowly through the "Force Region" entity.

**Actual behavior**
"Rigid Body" gets launched upwards by the "Force Region" entity.

**Video**

https://user-images.githubusercontent.com/86952082/216341751-f21416b3-e64b-4646-8936-d21351c120fb.mp4

**Found in Branch**
Development (e12b937)

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
e12b9375f881ae3b8b8485d3919ce77e64a70209

**Desktop**
 - Device: PC
 - OS: Windows
 - Version 11
 - CPU AMD Ryzen 5 3600
 - GPU NVIDIA GeForce RTX 2060 SUPER
 - Memory 16GB

**Comment by moraaar** [comment]
I've triaged this bug and I was able to reproduce it, any value bigger than 60 in the density of the "Simple Drag" force region will cause the box to shoot out.

The interesting thing is that behavior is actually acknowledged in the code in a comment but it's not shown to the user:
````
static const float s_forceRegionMaxDensity = 400.0f; // Large values create an oscillation that sends the body too far out. Legacy renderer's Octree may throw errors. Maximum density is defined as a value capable of slowing down a radius 1 ball weighing 1 ton.
````

Talking to @greerdv he suggested to introduce some code to clamp the force to avoid unstable behaviour, maybe clamp the force to stay under a value which would cause the object's velocity to drop to 0 in 1 second.

I can't take on this bug at this time, so I'll unassign myself so other people can take it if they can. Cheers.
